kolibri/plugins/coach/assets/test/views/reports/ReportsExerciseLearners.spec.js

```diff
@@ -0,0 +1,163 @@
+    statusObj: {
+      learner_id: 'd4b',
+      content_id: 'a97',
+      status: STATUSES.completed,
+      last_activity: new Date('2019-05-05T07:13:22Z'),
+      time_spent: 92.5,
+    },
+  },
+  {
+    id: 'a5d',
+    name: 'learner2',
+    username: 'learner2',
+    groups: ['group2'],
+    exerciseLearnerLink: '#/2e3/reports/lessons/79b/exercises/a97/learners/a5d',
+    statusObj: {
+      learner_id: 'a5d',
+      content_id: 'a97',
+      status: STATUSES.started,
+      last_activity: new Date('2019-05-05T08:13:22Z'),
+      time_spent: 17.14,
+    },
+  },
+];
+
+const initWrapper = propsData => {
+  return mount(ReportsExerciseLearners, {
+    propsData,
+    stubs: ['router-link'],
+  });
+};
+
+const getCol = (row, colIndex) => {
+  return row.findAll('td').at(colIndex);
+};
+
+describe('ReportsExerciseLearners', () => {
+  it('smoke test', () => {
+    const wrapper = shallowMount(ReportsExerciseLearners);
+
+    expect(wrapper.isVueInstance()).toBe(true);
+  });
+
+  it('renders all entries', () => {
+    const wrapper = initWrapper({
+      entries,
+    });
+
+    const rows = wrapper.findAll({ ref: 'entry' });
+    expect(rows.length).toBe(2);
+
+    const firstRow = rows.at(0);
+    const secondRow = rows.at(1);
+
+    expect(getCol(firstRow, 0).html()).toContain('learner1');
+    expect(getCol(firstRow, 1).html()).toContain(STATUSES.completed);
+    expect(getCol(firstRow, 2).html()).toContain('92 seconds');
+    expect(getCol(firstRow, 3).html()).toContain('group1, group2');
+    expect(getCol(firstRow, 4).html()).toContain(' hours ago');
+
+    expect(getCol(secondRow, 0).html()).toContain('learner2');
+    expect(getCol(secondRow, 1).html()).toContain(STATUSES.started);
+    expect(getCol(secondRow, 2).html()).toContain('17 seconds');
+    expect(getCol(secondRow, 3).html()).toContain('group2');
+    expect(getCol(secondRow, 4).html()).toContain(' hours ago');
+  });
+
+  it("doesn't render groups information when show groups set to false", () => {
+    const wrapper = initWrapper({
+      entries,
+      showGroupsColumn: false,
+    });
+
+    expect(wrapper.html()).not.toContain('Groups');
+    expect(wrapper.html()).not.toContain('group1');
+    expect(wrapper.html()).not.toContain('group2');
+  });
+
+  describe('when an exercise has started', () => {
+    let row;
+
+    beforeAll(() => {
+      const wrapper = initWrapper({
+        entries: [
+          {
+            name: 'learner1',
+            groups: [],
+            exerciseLearnerLink: '#/2e3/reports/lessons/79b/exercises/a97/learners/d4b',
+            statusObj: {
+              status: STATUSES.completed,
+              last_activity: new Date('2019-05-05T08:13:22Z'),
+              time_spent: 17.14,
+            },
+          },
+        ],
+      });
+
+      row = wrapper.find({ ref: 'entry' });
+    });
+
+    it("renders learner's name as a link to an exercise", () => {
+      expect(getCol(row, 0).html()).toContain('router-link');
```

Comment:
You can actually stub KRouterLink so it can be treated as a very simple vue component, and the Wrapper API works on it.

Here's an example https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/coach/assets/src/views/common/notifications/__test__/NotificationCard.spec.js#L15

Reply:
Not sure if I can understand this comment - why should I stub it here? I just want to check that the link is there.

Reply:
It's just an unusual test, because the actual rendered html will never have a `<router-link>` tag.

A test like
```
expect(getCol(row, 0).find({ name: 'KRouterLink' }).props().to).toEqual(...)
```

Mirrors the implementation a little more closely.

Reply:
Ah, ok, get what you meant now. Yep makes sense.